codec/codec_test.go

```diff
@@ -79,6 +81,63 @@ func TestMetrics(t *testing.T) {
 	}, metric, metricdatatest.IgnoreTimestamp())
 }
 
+func BenchmarkEncode(b *testing.B) {
+	ev := fullEvent(b)
+	testCases := map[string]struct {
+		codec Codec
+	}{
+		"json": {
+			codec: JSON{},
+		},
+		"vtproto": {
+			codec: VTProto{},
+		},
+	}
+	for name, tc := range testCases {
+		b.Run("format="+name, func(b *testing.B) {
```

Comment:
```suggestion
		b.Run(name, func(b *testing.B) {
```

Reply:
This was to make the stats comparison a bit easier. I can generate one file and compare the two codecs result by using the keys:
```
benchstat -col /format codec.txt
```

The reference is here:  reference is [here](https://pkg.go.dev/golang.org/x/perf/cmd/benchstat#hdr-Configuring_comparisons)